### PR TITLE
refactor: Simplify Schelling code 

### PR DESCRIPTION
**Pull request by rht**
1. Remove unused model attributes
2. Make `similar` calculation more natural language readable

With these changes and `shuffle_do`, the Schelling code in https://github.com/JuliaDynamics/ABMFrameworksComparison should go down from 44 to ~33, closer to Agents.jl's 28.

### CODE REVIEW

File: examples/schelling/model.py [hunk 1]
```diff
@@ -6,24 +6,21 @@ class SchellingAgent(mesa.Agent):
     Schelling segregation agent
     """
 
-    def __init__(self, model, agent_type):
+    def __init__(self, model: mesa.Model, agent_type: int) -> None:
         """
         Create a new Schelling agent.
```
Comment (rht) [inline examples/schelling/model.py:11]: This docstring is self explanatory for anyone knowing what `__init__` does, and should be considered a superfluous documentation.

Comment (rht) [reply examples/schelling/model.py:11]: I have read several articles about superfluous documentation, but couldn't find a good reference that explains it. Maybe this [SO thread](https://stackoverflow.com/questions/33019955/should-you-always-document-functions-even-if-redundant-specifically-python) does it.

File: examples/schelling/model.py [hunk 2]
```diff
@@ -95,5 +88,4 @@ def step(self):
 
         self.datacollector.collect(self)
 
-        if self.happy == len(self.agents):
-            self.running = False
+        self.running = self.happy != len(self.agents)
```
Comment (EwoutH) [inline examples/schelling/model.py:91]: While this is fully correct, I find it less readable. Maybe this could also be resolved with a comment.

**Review by EwoutH — APPROVED** [review]
Thanks, especially the step simplification is really nice.

One comment, otherwise good. Maybe check with @quaquel if it doesn’t conflict his work.

**Comment by quaquel** [comment]
It does conflict with #198 but I can easily rectify that there once this is merged. 

It would be good to move these changes also to the benchmarks in the main repo (is this an argument in favor of [#2349](https://github.com/projectmesa/mesa/pull/2349)? 😉 )

File: examples/schelling/model.py (hunk 2 above)
Comment (quaquel) [reply examples/schelling/model.py:91]: My experience is the other way around. I found this very clear.

**Review by quaquel — APPROVED** [review]

**Pull request merged**